Add debug config migration guide for RNT preview (#1311)
microsoft/vscode-react-native: README.md

> This is a preview version of the updated React Native Tools extension. It's not recommended to use the preview version and the stable one at the same time.

## React Native Tools Preview migration guide

> In the Preview version of the extension we changed debug types in debug configurations. The table below shows the difference between debug configuration of the stable and the preview version.

Reviewer: ```suggestion
In the Preview version of the extension we changed debug types in debug configurations. The table below shows the difference between the debug configurations of the stable and the preview version.
```

Reviewer 2: I think we could just cut this top part all together. I think this line:
> In order to use the preview version of the extension it's required to add `-preview` suffix in `type` field of a debug configuration of the RNT stable extension.

Explains exactly the changes, and also what users need to do to switch their launch configs.

> |RNT stable|RNT preview|

Reviewer 3: I am also thinking that we could add some image to demonstrate how the old configuration becomes the preview one in addition to this table.

|---|---|

> |```"type": "reactnative"```<br>![Android debug config stable](images/debug-config-android-stable.png)|```"type": "reactnative-preview"```<br>![Android debug config preview](images/debug-config-android-preview.png)|

Reviewer: I think we can use ` instead of ``` here. Or are there any rendering issues with tables and HTML?

Reviewer 2: I think we could also cut this table and just use a diff block to show users what they need to change, e.g.:

## Web debugging:
```diff
{
    "name": "Debug Android",
    "cwd": "${workspaceFolder}",
-    "type": "reactnative",
+    "type": "reactnative-preview",
    "request": "launch",
    "platform": "android"
}
```
## Direct debugging:
```diff
{
    "name": "Debug Android",
    "cwd": "${workspaceFolder}",
-    "type": "reactnativedirect",
+    "type": "reactnativedirect-preview",
    "request": "launch",
    "platform": "android"
}
```

|```"type": "reactnativedirect"```|```"type": "reactnativedirect-preview"```|

> It's just required to add `-preview` suffix in `type` field of a debug configuration of the RNT stable extension to be able to use the preview version of the extension.

Reviewer 3: ```suggestion
In order to use the preview version of the extension it's required to add `-preview` suffix in `type` field of a debug configuration of the RNT stable extension.
```

We also added `(Preview)` suffix to Command Palette command names.

![React Native commands preview](images/command-palette-preview.png)

> To avoid conflicts, it's not recommended to use Command Palette commands of the preview version and the stable one at the same time.

Reviewer 3: ```suggestion
To avoid conflicts, it's not recommended to use Command Palette commands and debug configurations of the preview version and the stable one at the same time.
```

> In order to use the preview version of the extension it's required to add `-preview` suffix in `type` field of a debug configuration of the RNT stable extension.

Reviewer: ```suggestion
In order to use the preview version of the extension it's required to add `-preview` suffix in the `type` field of a debug configuration of the RNT stable extension.
```
